### DISCUSSION TITLE
Can I pass specific function for a particular step?

### DISCUSSION THREAD

**Discussion by tazwar9t63** [gilbarbara/react-joyride#724] [Q&A] [answered]
I want to expand checkbox for a step and I already have a function for it, is this possible to execute the function for that step? and I want to implement it for multiple steps, I've got the index of each step as well
@gilbarbara

**Comment by gilbarbara** [comment, marked as answer]
hey @tazwar9t63 

The Step typing doesn't support custom properties at the moment.
But if you know the step's index, you can call your function in the [callback](https://docs.react-joyride.com/callback).

**Reply by tazwar9t63** [reply]
I've already tried using callback function like this 

`const handleJoyrideCallback = (callbackData) => {
    const { index } = callbackData;
    console.log({ index });
return index === 0 ? authorLabelClick() : index === 1 ? sectionLabelClick() : index === 2 ? dateLabelClick() : "";
};
`
But it doesn't work properly
PS: I have three steps

**Reply by gilbarbara** [reply]
> But it doesn't work properly

Why not?
Try to trigger your function when the `callbackData.type === 'tooltip'`
